d3a_api_client/setups/test_ess_bid_redis_aggregator.py

```diff
@@ -19,17 +20,15 @@ def on_market_cycle(self, market_info):
             if "energy_to_buy" in device_event["device_info"] and \
                     device_event["device_info"]["energy_to_buy"] > 0.0:
                 buy_energy = device_event["device_info"]["energy_to_buy"] / 2
-                batch_commands[device_event["area_uuid"]] = [
-                    {"type": "bid",
-                     "price": 31 * buy_energy,
-                     "energy": buy_energy},
-                    {"type": "list_bids"}]
-
-        logging.info(f"batch_commands: {batch_commands}")
-
-        if batch_commands:
-            self.batch_command(batch_commands)
-            logging.info(f"Batch command placed on the new market")
+                self.add_to_batch_commands.bid_energy(area_uuid=device_event["area_uuid"],
+                                                      price=31 * buy_energy, energy=buy_energy).\
+                    list_bids(area_uuid=device_event["area_uuid"])
+                self.is_buffer_empty = False
+
+        if not self.is_buffer_empty:
+            response = self.execute_batch_commands(batch_commands)
+            self.is_buffer_empty = True
+            logging.info(f"Batch command placed on the new market: {response}")
```

Comment:
Can you please explain the reason for this (self.is_buffer_empty)?

Reply:
This is just to check if the batch commands buffer is empty, anyway if it is empty and we called the execute_batch_commands ,it will not execute anything so no harm in both ways

Reply:
The `RedisAggregator` has a `_client_command_buffer` attribute of type `ClientCommandBuffer`. This class comes with a `buffer_length` property: what would you think of using that to check whether the buffer is empty or not?

https://github.com/gridsingularity/d3a-api-client/blob/e8c51844decf376c93a037f7aea636a27a71a2da/d3a_api_client/commands.py#L9-L11

Maybe you could even define it as a property shortcut on the `RedisAggregator`.

Reply:
This is the user facing side, so I suggest to remove the check if it is not needed, Otherwise there will be questions.